types/Extrinsic.ts

```diff
@@ -30,77 +30,83 @@ export interface ExtrinsicValueV4 {
  * It is lightly modified [[ExtrinsicV4]] from `@polkadot/types`
  */
 export default class PlugExtrinsicV1 extends Struct implements IExtrinsicImpl {
-  public constructor (value?: Uint8Array | ExtrinsicValueV4 | Call, { isSigned }: Partial<ExtrinsicOptions> = {}) {
-    super({
-      signature: PlugExtrinsicSignatureV1,
-      method: 'Call'
-    }, PlugExtrinsicV1.decodeExtrinsic(value, isSigned));
-  }
-
-  public static decodeExtrinsic (value?: Call | Uint8Array | ExtrinsicValueV4, isSigned = false): ExtrinsicValueV4 {
-    if (!value) {
-      return {};
-    } else if (value instanceof PlugExtrinsicV1) {
-      return value;
-    } else if (value instanceof ClassOf('Call')) {
-      return { method: value };
-    } else if (isU8a(value)) {
-      // here we decode manually since we need to pull through the version information
-      const signature = new PlugExtrinsicSignatureV1(value, { isSigned });
-      const method = createType('Call', value.subarray(signature.encodedLength));
-
-      return {
-        method,
-        signature
-      };
+    /**
```

Comment:
Could we place these in a separated PR, and link a task ticket if possible, so that we all know the ticket background and I am able to give a proper review opinion? What do you think @holygits